rosidl_runtime_py/set_message.py

```diff
@@ -54,6 +55,8 @@ def set_message_fields(
     def set_message_fields_internal(
             msg: Any, values: Dict[str, str],
             timestamp_fields: List[Any]) -> List[Any]:
+        # Create a deep copy of the input dictionary to avoid modifying it for safety
+        values = copy.deepcopy(values)
```

Comment:
Since this function handles the nested structure internally, i would use `deepcopy` for the safety.

Reply:
Does this have any sort of performance impact?

Reply:
of course it does. but i think this is not performance sensitive API. (besides python...)